### ISSUE TITLE
Saving session

### ISSUE THREAD

**Issue by glennfu** [turbolinks/turbolinks-ios#30]
Has any thought/effort happened so far toward saving sessions? It would be great if loading the app later didn't log you out. I tried implementing this myself but it seems that with WKWebView, managing cookies is a lot trickier than with the old UIWebView. This is the only online reference I found to anyone saying they were able to do it successfully: http://stackoverflow.com/a/32845148/1366620

Is saving session in the scope of the turbolinks-ios project? If so, is this kind of solution the right approach for making this work, or is something cleaner possible? I know I'm jumping on board the train before it's left the station so you might all have plans in place for this already, especially since I know it works in Basecamp 3, but I thought I'd share what I've found and ask!

**Comment by joemasilotti** [comment]
I had a similar discussion with @sstephenson in PR #20.

**tl;dr**: Use a persistent cookie on the server.

For example, in the demo app replace `app.rb`'s `sign-in` method with the following (from [this comment](https://github.com/turbolinks/turbolinks-ios/pull/20#issuecomment-201964497):

```ruby
post '/sign-in' do
  response.set_cookie(:signed_in, value: true, expires: Date.parse('2020-01-01').to_time)
  redirect to('/')
end
```

We are using this in our app and it's working beautifully.

**Comment by zachwaugh** [comment]
As @joemasilotti mentioned, cookies should be handled automatically if configured correctly. If you're authenticating with a web view, and all your requests are made from the web view, that should be all you need to do.

If you need to also make native API calls (with `NSURLSession`, etc), it's not quite as straightforward. For the Basecamp 3 app, we chose to use a combination of OAuth and cookies. We authenticate initially with OAuth, and use OAuth tokens for any native API requests. We then use that OAuth token to authenticate the web view and have the server to set the correct session cookies, which are automatically persisted. This avoids various hacky methods of getting/setting cookies manually in the web view.

**Comment by glennfu** [comment]
"Use a persistent cookie on the server." - I can't believe I wasn't doing that. Sorry all, and thank you!

**Comment by imweijia** [comment]
This is absolutely gold. I almost went down a complicated path of handing it in WKWebView and a javascript bridge. This solves everything!

**Comment by vitormil** [comment]
My **turbolinks-android** app is working fine, but I'm still facing this issue in my **turbolinks-ios** app.

Could be a lack of permission? Any thoughts?

**Comment by shobhakartiwari** [comment]
Hi , in my WebView login session maintained and working perfectly by using HTTPCookies storage but when i switched to WKWebView , I really find it difficult and login session expired , i dont find any link helpful for handling this in WKWebview as i am handling in Swift 3 ( Xcode 8.3.3) so https://stackoverflow.com/questions/26573137/can-i-set-the-cookies-to-be-used-by-a-wkwebview/32845148#32845148 doesn't work . is there any workaround , i really need this . I appreciate if any1 share such code or any link in swift for handling this?
